torch/utils/benchmark/utils/common.py

```diff
@@ -288,3 +293,63 @@ def set_torch_threads(n: int) -> Iterator[None]:
         yield
     finally:
         torch.set_num_threads(prior_num_threads)
+
+
+def _make_temp_dir(prefix: Optional[str] = None, gc_dev_shm: bool = False) -> str:
+    """Create a temporary directory. The caller is responsible for cleanup.
+
+    This function is conceptually similar to `tempfile.mkdtemp`, but with
+    the key additional feature that it will use shared memory if the
+    `BENCHMARK_USE_DEV_SHM` environment variable is set. This is an
```

Comment:
Why use environment variable to control the function behavior? Is it because it is hard to pass the parameters to this level?

Reply:
Mainly because it's meant to be an environment level switch. (It only really matters when you're beating the hell out of a machine.)